corebehrt/main_causal/helper/calibrate_exp_y.py

```
@@ -0,0 +1,234 @@
+    calibrated_probas: np.ndarray,
+    preds: np.ndarray,
+    epsilon: float = 1e-8,
+    collapse_threshold: float = CALIBRATION_COLLAPSE_THRESHOLD,
+) -> np.ndarray:
+    """
+    If the calibrated probabilities appear to be collapsed (have very low variance),
+    keeps the original probabilities instead. Returns the updated validation dataframe.
+    """
+    calibrated_probas = np.clip(calibrated_probas, epsilon, 1 - epsilon)
+    if np.std(calibrated_probas) < collapse_threshold:
+        warnings.warn(
```

Ruff (0.11.9)

No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)